### DISCUSSION TITLE
file names

### DISCUSSION THREAD

**Discussion by reymalahay** [bxparks/AUnit#115] [General]
hi all,

if any of you use the `#line 2` syntax in your tests, i.e.
```
#line 2 yourfilename.ino
```

if you use special characters in your `.ino` file, like underscores, e.g. `this_is_my_file.ino`, you will get an error like the one listed below:
```
"this_is_my_file" is not a valid filename 
```
and your code will not compile.

a workaround for this issue is to escape the `this_is_my_file.ino` filename with double quotes, i.e.:
```
#line 2 "yourfilename.ino"
```

then the error goes away and your code will compile successfully.

kind regards,
rey malahay

**Comment by bxparks** [comment]
The `#line` directive takes a string constant:
```
#line digit-sequence ["filename"]
```
(see for example, https://learn.microsoft.com/en-us/cpp/preprocessor/hash-line-directive-c-cpp), and I verified that all my unit test code everywhere uses the double-quotes, so if people just copy my examples, they should be fine.

There might be some potential for confusion when compared to the `#include` directive, which can use the `<filename>` syntax, but even there, double-quotes are required for the other syntax, `#include "filename"`.

**Reply by bxparks** [reply]
I noticed that I did not use proper double-quotes in 2 places in the README.md which mention the `#line` statement. Fixed in commit 15b8fde503226eed88dcc079c6be21ff338372ce.

**Comment by bxparks** [comment]
I should also remark that this `#line 2 ["filename"]` has always bugged me. It shouldn't be necessary, but some compilers get the line number of the error message wrong for reasons which are completely mysterious to me. I also seem to recall that some compilers seem to have fixed something internal, so that they print the correct line number without this hack. I have never taken the time to methodically track down which combinations of (compiler, version, microcontroller, environment) have fixed this problem, which have not.
